### PR TITLE
TDD methods

### PR DESCRIPTION
**Pull request by thecodechaser**
## This branch includes the following features:
-  We implemented a class with some functions but we did it in Test-driven-development(TDD).
-  Implemented a class Solver
-  Implemented methods fizzbuzz, factorial, and reverse.
-  Implemented unit tests for methods fizzbuzz, factorial, and reverse.

### CODE REVIEW

File: spec/factorial_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      expect(result).to eq 24
+    end
+
+    it 'the method factorial should return the factorial of given integer' do
+      value = 6
+      result = solver.factorial(value)
+      expect(result).to eq 720
+    end
+  end
+end
```
Comment (Mihndim2020) [inline spec/factorial_spec.rb:36]: - [ ] Since it is a requirement for the this project and best practice, testing edge cases, it will be nice to add a test case which handles a `negative value and raises an exception as an expectation`
- [ ] It will also be nice to add tests for special cases of 0 and 1.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi Team, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi team,

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
